src/aoc2020/day1.clj

```diff
@@ -0,0 +1,143 @@
+(ns aoc2020.day1
+  (:require [aoc2020.util :refer [file->seq]]))
+
+
+;; part 1
+
+(defn strs->ints [strs]
+  "parsing strings to integers"
+  (let [str->int (fn [str]
+                   (Integer/parseInt str))]
+    (map str->int strs)))
+
+(defn int->compl [n sum]
+  "calculate complement"
+  (- sum n))
+
+(defn compl-in-set? [n sum s]
+  "detect whether the complement is in the sequence (using set)"
+  (let [compl (int->compl n sum)]                           ; calculate complement
+    (contains? s compl)))                                   ; check if exist in set
+
+(defn int->int*compl [n sum]
+  "multiply x and its complement"
+  (if n                                                     ; if n is not nil
```

Comment:
See `when`

Reply:
because in three-sum problem, we need to run the two-sum for every number, but it may not have the valid complement pair for some numbers